include/invert_laplace.hxx

```diff
@@ -132,17 +135,19 @@ constexpr int INVERT_KX_ZERO = 16;
 class LaplaceFactory
     : public Factory<
           Laplacian, LaplaceFactory,
-          std::function<std::unique_ptr<Laplacian>(Options*, CELL_LOC, Mesh*)>> {
+          std::function<std::unique_ptr<Laplacian>(Options*, CELL_LOC, Mesh*, Solver*,
+                                                   Datafile*)>> {
 public:
   static constexpr auto type_name = "Laplacian";
   static constexpr auto section_name = "laplace";
   static constexpr auto option_name = "type";
   static constexpr auto default_type = LAPLACE_CYCLIC;
 
   ReturnType create(Options* options = nullptr, CELL_LOC loc = CELL_CENTRE,
-                    Mesh* mesh = nullptr) {
+                    Mesh* mesh = nullptr, Solver* solver = nullptr,
+                    Datafile* dump = nullptr) {
```

Comment:
Just a note that when we get the output refactoring in, this will be a bit nicer as we'd be able to drop the `Datafile`